README.md

```diff
@@ -145,12 +145,85 @@ contains a copy of it's own release notes.
 
 We should also copy the release notes to the Github releases page, but CHANGELOG.md is the primary place to keep the release notes. 
 
-The release commit should be tagged with a signed tag:
+The release commit should be tagged with an annotated and signed tag:
 
-    git tag -s vx.x.x
+    git tag -asm vx.x.x vx.x.x
     git push --tags
 
+A non-prescriptive example of the release process might look like the following:
 
+```sh
+git checkout master
+git pull                                # checkout/pull latest master
+git checkout -b vx.x.x-release          # create release branch
+vi CHANGELOG.md                         # update CHANGELOG.md
+make visor                              # validate build
+go mod tidy                             # ensure tidy go.mod for release
```

Comment:
tidy should be done before build so the build uses the exact same versions as specified in go.mod/go.sum. Ideally go build should be run with `-mod=readonly` so it doesn't change the files. This is the new default in the upcoming version of Go.